src/processor/queue-consumer.service.ts

```diff
@@ -197,8 +331,44 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
       } else {
         await this.eventEmitter.emitAsync('capacity.refilled');
       }
-    } catch (err) {
-      this.logger.error('Caught error in checkCapacity', err);
+    } catch (err: any) {
+      this.logger.error('Caught error in checkCapacity', err?.stack);
     }
   }
+
+  private handleMessagesFailure(error: string): { pause: boolean; retry: boolean } {
+    let retval = { pause: false, retry: false };
+    switch (error) {
+      case 'StalePageState':
+      case 'ProofHasExpired':
+      case 'ProofNotYetValid':
+      case 'InvalidSignature':
```

Comment:
Isn't `InvalidSignature` also non-recoverable?

Reply:
Hmm, good point... this was an existing method I just moved here from another file, but yeah.

Reply:
Actually, `InvalidSignature` maybe should pause the queue, since it probably indicates a global key problem with the provider seed phrase... it's unlikely that we'd have a signature that didn't match the payload, but more likely that the signing keys are invalid. Thoughts @wilwade ?

Reply:
It would mean that one user has a bad key. Not the capacity tx key.

Reply:
No; all transactions are signed by the provider keypair only; nothing is "signed" by the graph key, that's only for encryption. So `InvalidSignature` can only mean either a signature/payload mismatch, or a bad provider signing key, right?

Reply:
Ah right. This is not a signature required tx. I think in that case it is not possible for this error to occur at all. `InvalidSignature` only happens when it is an invalid user signature.